library/src/include/auxiliary.hpp

```diff
@@ -214,7 +214,7 @@ HIPBLASLT_EXPORT
 constexpr hipDataType string_to_hip_datatype_assert(const std::string& value)
 {
     auto datatype = string_to_hip_datatype(value);
-    if(static_cast<int>(datatype) == 0)
+    if(datatype == HIPBLASLT_DATATYPE_INVALID)
```

Comment:
The corresponding value for 'static_cast<int>(datatype) == 0' is HIP_R_32F. To verify that it is an invalid datatype, it should be compared with [HIPBLASLT_DATATYPE_INVALID](https://github.com/ROCm/hipBLASLt/pull/1205/files#diff-70872ce09b5dadefe21a46261af09f58ee26523be88e13b3768612cab29a681fR210).